### DISCUSSION TITLE
Augment the AuthUser object?

### DISCUSSION THREAD

**Discussion by jamie-legg** [gladly-team/next-firebase-auth#481] [Q&A]
Is it possible to augment the `AuthUser` object (perhaps by modifying `req` before  `setAuthCookies(req, res)` in order to store additional authorisation data? For example, the role someone has in an admin dashboard.

Forgive me if this is explored elsewhere, I did look.

I want to do something like, verify the token server side, use `admin` to get the role of the operator, and then store that in the cookie for easy retrieval throughout the application components.

**Comment by kmjennison** [comment]
Could you use custom Firebase claims for this, then rely on the `AuthUser.claims` value?

**Reply by jamie-legg** [reply]
I'll explore in that direction and see how it goes. I'm relying on claims for user permissions at the moment, and trying to keep them somewhat separate from admin permissions
